src/scval.js

```diff
@@ -82,9 +107,35 @@ import { ScInt, scValToBigInt } from './numbers/index';
  *    types, custom classes)
  *  - the type of the input object (or some inner value of said object) cannot
  *    be determined (via `typeof`)
+ *  - the type you specified (via `opts.type`) is incompatible with the value
```

Comment:
it only throws for vals that have sub-types that actually check on 'type' , non-sub-types like address, contract, object, will just ignore, not sure if this inconsistency is important.

Reply:
Great point! Done in 2b1b010b.